realm/realm-library/src/objectServer/java/io/realm/SyncManager.java

```diff
@@ -230,6 +232,25 @@ private static synchronized void removeSession(SyncConfiguration syncConfigurati
         }
     }
 
+    /**
+     * Retruns the all valid sessions belonging to the user.
```

Comment:
Right now the Object Store controls the session lifecycle, which is actually the biggest pain around sessions. We need to decouple this down the line for other functionality like start/stopping it.

This was the primary reason I said to @dalinaum that just using the Java map was fine. Maintaining two Session objects (in SyncManager and this one) will surely also cause a lot of problems. Ideally, we would only have one Java object representing the session.